channels/rdpei/client/rdpei_main.c

```diff
@@ -410,18 +428,58 @@ static UINT rdpei_add_pen_frame(RdpeiClientContext* context)
 	return CHANNEL_RC_OK;
 }
 
-static UINT rdpei_update(RdpeiClientContext* context)
+static UINT rdpei_update(wLog* log, RdpeiClientContext* context)
 {
 	UINT error = rdpei_add_frame(context);
 	if (error != CHANNEL_RC_OK)
 	{
-		WLog_ERR(TAG, "rdpei_add_frame failed with error %" PRIu32 "!", error);
+		WLog_Print(log, WLOG_ERROR, "rdpei_add_frame failed with error %" PRIu32 "!", error);
 		return error;
 	}
 
 	return rdpei_add_pen_frame(context);
 }
 
+static BOOL rdpei_poll_run_unlocked(rdpContext* context, void* userdata)
```

Comment:
Actually I'd just have `rdpei_poll_run` and protect the call in the thread with `Enter/LeaveCriticalSection`.
Making two versions with _lock, _unlock or _nothing_ just increases the chance of picking the wrong one (as is currently done in the `freerdp_client_channel_register` call).

Reply:
@mfleisz why do you think locking is not required for the `freerdp_client_channel_register` call?

Reply:
@akallabeth sorry - thought the CS is only required due to the thread created in the channel - but of course it's not! 😅